subprojects/packagefiles/luajit/src/meson.build

```diff
@@ -0,0 +1,274 @@
+    'lj_api.c',
+    'lj_asm.c',
+    'lj_assert.c',
+    'lj_bc.c',
+    'lj_bcread.c',
+    'lj_bcwrite.c',
+    'lj_buf.c',
+    'lj_carith.c',
+    'lj_ccall.c',
+    'lj_ccallback.c',
+    'lj_cconv.c',
+    'lj_cdata.c',
+    'lj_char.c',
+    'lj_clib.c',
+    'lj_cparse.c',
+    'lj_crecord.c',
+    'lj_ctype.c',
+    'lj_debug.c',
+    'lj_dispatch.c',
+    'lj_err.c',
+    'lj_ffrecord.c',
+    'lj_func.c',
+    'lj_gc.c',
+    'lj_gdbjit.c',
+    'lj_ir.c',
+    'lj_lex.c',
+    'lj_lib.c',
+    'lj_load.c',
+    'lj_mcode.c',
+    'lj_meta.c',
+    'lj_obj.c',
+    'lj_opt_dce.c',
+    'lj_opt_fold.c',
+    'lj_opt_loop.c',
+    'lj_opt_mem.c',
+    'lj_opt_narrow.c',
+    'lj_opt_sink.c',
+    'lj_opt_split.c',
+    'lj_parse.c',
+    'lj_prng.c',
+    'lj_profile.c',
+    'lj_record.c',
+    'lj_serialize.c',
+    'lj_snap.c',
+    'lj_state.c',
+    'lj_str.c',
+    'lj_strfmt.c',
+    'lj_strfmt_num.c',
+    'lj_strscan.c',
+    'lj_tab.c',
+    'lj_trace.c',
+    'lj_udata.c',
+    'lj_vmevent.c',
+    'lj_vmmath.c',
+  ]
+endif
+
+luajit_lib = library(
+  'luajit-' + abiver,
+  luajit_lib_src,
+  c_args: luajit_lib_flags,
+  dependencies: [m_dep, dl_dep],
+  install: true,
+  override_options: ['b_sanitize=none'],
+  soversion: meson.project_version(),
+  # A version number like `2.1.1713773202` breaks clang when compiling for x86_64 on macOS:
+  # ```
+  # cc  -o […]/libluajit-5.1.2.1.1713773202.dylib […] -compatibility_version 2.1.1713773202 -current_version 2.1.1713773202 -lm -ldl
+  # ld: malformed 32-bit x.y.z version number: 2.1.1713773202
+  # clang: error: linker command failed with exit code 1 (use -v to see invocation)
+  # ````
+  # So use the same versions as in the official makefile.
+  darwin_versions: [ljver, ljver + '.255'],
+)
+
+luajit_dep = declare_dependency(
+  include_directories: include_directories('.'),
+  link_with: luajit_lib,
+  sources: luajit_h,
+  variables: {
+    'INSTALL_CMOD': install_cmod,
+    'INSTALL_LMOD': install_lmod,
+  },
+)
+meson.override_dependency('luajit', luajit_dep)
+
+if get_option('default_library') == 'static'
+  if host_machine.system() == 'windows'
+    error('building a static LuaJIT library is not supported on Windows')
+  endif
+  luajit_exe_export_dynamic = true
+else
+  luajit_exe_export_dynamic = false
+endif
+
+luajit_exe = executable(
+  'luajit-' + meson.project_version(),
+  [luajit_h, 'luajit.c'],
+  export_dynamic: luajit_exe_export_dynamic,
+  install: get_option('luajit'),
+  link_with: luajit_lib,
+)
+meson.override_find_program('luajit', luajit_exe)
+
+luajit_exe_basename = fs.name(luajit_exe.full_path())
+luajit_exe_extension = luajit_exe_basename.split(luajit_exe.name())[1]
+
+install_symlink(
+  'luajit' + luajit_exe_extension,
+  pointing_to: luajit_exe_basename,
+  install_dir: get_option('bindir'),
+  install_tag: 'runtime',
+)
+
+install_data(
+  '../COPYRIGHT',
+  install_dir: get_option('datadir') / 'licenses/luajit',
+  install_tag: 'doc',
```

Comment:
This can be handled by the license_files argument to project() albeit raises the meson version requirement.

Benefit being that it's tied into the builtin licensedir option.

Reply:
`1.1.0`, maybe a little to fresh?

Reply:
I tweaked the code to use `licensedir` on `meson >= 1.1.0` instead.